### PR TITLE
[REVIEW] MG BFS Dask PR

### PR DESCRIPTION
**Pull request by kaatish**

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cugraph/opg/traversal/mg_bfs_wrapper.pyx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    cdef size_t handle_size_t = <size_t>handle.getHandle()
+    handle_ = <c_bfs.handle_t*>handle_size_t
+
+    # Local COO information
+    src = input_df['src']
+    dst = input_df['dst']
+    num_verts = local_data['verts'].sum()
+    num_edges = local_data['edges'].sum()
+    local_offset = local_data['offsets'][rank]
+    dst = dst - local_offset
```
Comment (Iroy30) [inline python/cugraph/opg/traversal/mg_bfs_wrapper.pyx:38]: probably want to do src = src - local_offset

File: python/cugraph/opg/traversal/mg_bfs_wrapper.pyx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    cdef uintptr_t c_predecessor_ptr = df['predecessor'].__cuda_array_interface__['data'][0];
+    if (return_distances):
+        c_distance_ptr = df['distance'].__cuda_array_interface__['data'][0]
+
+    # Extract local data
+    cdef uintptr_t c_local_verts = local_data['verts'].__array_interface__['data'][0]
+    cdef uintptr_t c_local_edges = local_data['edges'].__array_interface__['data'][0]
+    cdef uintptr_t c_local_offsets = local_data['offsets'].__array_interface__['data'][0]
+    if (return_distances):
+        df['distance'] = cudf.Series(np.zeros(num_verts, dtype=np.int32))
```
Comment (Iroy30) [inline python/cugraph/opg/traversal/mg_bfs_wrapper.pyx:71]: Already initialized

File: python/cugraph/dask/bfs/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from .bfs import bfs, get_chunksize, read_split_csv, drop_duplicates
```
Comment (Iroy30) [inline python/cugraph/dask/bfs/__init__.py:1]: This File is not needed. Pagerank is also going to be removed for 0.15. No point adding it to be soon removed

File: python/cugraph/dask/bfs/bfs.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+def bfs(edge_list, start, return_distances=False):
```
Comment (Iroy30) [inline python/cugraph/dask/bfs/bfs.py:1]: This File is not needed. Corresponding Pagerank counterpart is also going to be removed for 0.15. No point adding it to be soon removed

File: python/cugraph/dask/mg_bfs/bfs.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    return_sp_counter : bool, optional, default=False
+        Indicates if shortest path counters should be returned
+
+    Returns
+    -------
+    df : cudf.DataFrame
+        df['vertex'][i] gives the vertex id of the i'th vertex
+
+        df['distance'][i] gives the path distance for the i'th vertex from the
```
Comment (afender) [inline python/cugraph/dask/mg_bfs/bfs.py:62]: What happens to `df['distance'][i]`  when `return_distances=False`? this should be documented.

File: python/cugraph/tests/dask/test_mg_bfs.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+from dask_cuda import LocalCUDACluster
+
+
+def test_dask_pagerank():
+    gc.collect()
+    cluster = LocalCUDACluster()
+    client = Client(cluster)
+    Comms.initialize()
+
+    input_data_path = r"../datasets/karate.csv"
```
Comment (afender) [inline python/cugraph/tests/dask/test_mg_bfs.py:30]: only karate? we could add a bigger one like net science too.

**Pull request merged**